internal/scheduler/legacyscheduler.go

```diff
@@ -0,0 +1,399 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/G-Research/armada/internal/armada/configuration"
+	"github.com/G-Research/armada/internal/armada/scheduling"
+	"github.com/G-Research/armada/internal/common"
+	"github.com/G-Research/armada/internal/common/logging"
+	"github.com/G-Research/armada/internal/scheduler/schedulerobjects"
+	"github.com/G-Research/armada/pkg/api"
+	"github.com/G-Research/armada/pkg/armadaevents"
+)
+
+type LegacyScheduler struct {
```

Comment:
I think you're going to need some mutexes here (or make sure this code is marked as not threadsafe and get callers to use a mutex)